### ISSUE TITLE
Code linting

### ISSUE THREAD

**Issue by ejfrancis** [prosemeteor/prosemirror#16]
Integrate an eslint configuration, and pre-commit hooks to prevent committing code if it fails linting.

**Comment by ejfrancis** [comment]
This is the config that I usually use, which extends the [standard config](https://github.com/feross/eslint-config-standard). It's annoyingly strict at first, but leads to really uniform code and helps avoid some mistakes once you get used to it.

```
module.exports = {
  root: true,
  // https://github.com/feross/standard/blob/master/RULES.md#javascript-standard-style
  extends: 'standard',
  // add your custom rules here
  'rules': {
    //  always require callbacks to be returned
    'callback-return': 2,
    // allow paren-less arrow functions
    'arrow-parens': 0,
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0,
    // require semicolons
    'semi': [2, 'always'],
    // don't allow multiple empty lines
    'no-multiple-empty-lines': 0,
    // require a single space before function parens
    'space-before-function-paren': 0,
    // don't allow "var", only "let" or "const"
    'no-var':0
  }
}
```

**Comment by funkyeah** [comment]
@ejfrancis what text editor/IDE do you use? also do you have the linter to run after every save or what are the settings for how you run it?

**Comment by ejfrancis** [comment]
I'm using Atom at the moment, although I switch between that and WebStorm. I added the `eslint-watch` package so that when you run the demo it'll run linting once when you start it, and then every time you make an edit to a file it will re-run on that single file. I've been using the demo as a playground to test out changes so that's been working out alright.  Apart from that, I added a pre-commit and pre-push hook, so it should run linting every time you commit/push and prevent you from doing so if the linting fails.
